src/turbolinks/history.coffee

```diff
@@ -35,11 +30,18 @@ class Turbolinks.History
         restorationIdentifier = turbolinks.restorationIdentifier
         @delegate.historyPoppedToLocationWithRestorationIdentifier(location, restorationIdentifier)
 
+  onPageLoad: (event) =>
+    Turbolinks.defer =>
+      @pageLoaded = true
+
   # Private
 
   shouldHandlePopState: ->
     # Safari dispatches a popstate event after window's load event, ignore it
-    pageLoaded is true
+    @pageIsLoaded()
+
+  pageIsLoaded: ->
+    @pageLoaded or document.readyState is "complete"
```

Comment:
Added an additional check for `document.readyState` to cover the theoretical case where turbolinks is loaded async after the `load` event.